k8s/node.go

```diff
@@ -57,9 +57,10 @@ func GetNode(k8sClient *kubernetes.Clientset, nodeName, cloudProviderName string
 				return &node, nil
 			}
 			if cloudProviderName == ExternalAWSCloudProviderName {
-				logrus.Debugf("Checking hostname address for node [%v], cloud provider: %v", nodeName, cloudProviderName)
+				logrus.Debugf("Checking internal address for node [%v], cloud provider: %v", nodeName, cloudProviderName)
 				for _, addr := range node.Status.Addresses {
-					if addr.Type == v1.NodeHostName && strings.ToLower(node.Labels[HostnameLabel]) == addr.Address {
+					if addr.Type == v1.NodeInternalIP && nodeAddress == addr.Address {
```

Comment:
I agree with this. Logic in the caller function should output the err already if you return it from here. Otherwise lgtm

Reply:
@jiaqiluo Addressing both comments here since they're related. 
`nodeAddress` or `internalAddress` won't be empty because rke defaults to `address`/`externalIP` when it's empty in cluster/defaults. 
https://github.com/rancher/rke/blob/c56fb8c42a079d569dd402c1243a5ae16b4004c9/cluster/defaults.go#L225

rke sets `node-ip` for kubelet and kubelet kinda uses it to set the `internalIP` so `internalIP` shouldn't be empty either. 
https://github.com/rancher/rke/blob/c56fb8c42a079d569dd402c1243a5ae16b4004c9/cluster/plan.go#L495
https://github.com/kubernetes/kubernetes/blob/ad9b60e2c9ddb21e8b00cabbe27e639638a0ea88/pkg/kubelet/nodestatus/setters.go#L231 

I can still add check for empty fields check, but I don't have fallback field to find the node instead with. Other addresses like private dns and external ip get set by aws cloud controller manager, but since that's deployed after cluster provisions we only have internal ip and hostname to find the node against. So in this case, it'll error out with `NotFound` error. What do you think?

Reply:
@kinarashah, thank you for the explanation. I think we should return an error saying that the provided IP can not be empty, instead of NotFound.

Reply:
@jiaqiluo the provided IP or nodeAddress could come from user but could also be set by RKE if it's empty, but since it's crucial to finding node, I'll add a check to be safe.